upload error during stress testing

```
{"level":"info","ts":1696448527.64371,"kind":"server","component":"grpc","operation":"/cas.v1.StatusService/Statusz","args":"readiness:true","code":0,"reason":"","stack":"","latency":0.00006399}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa265a1]

goroutine 245650 [running]:
github.com/chainloop-dev/chainloop/app/artifact-cas/internal/service.bufferStream({0x12814f0, 0xc0004627b0}, {0x1286c38, 0xc00576c480}, 0xc005a42420, 0xc0001391a0)
        /home/runner/work/chainloop/chainloop/app/artifact-cas/internal/service/bytestream.go:196 +0x141
created by github.com/chainloop-dev/chainloop/app/artifact-cas/internal/service.(*ByteStreamService).Write in goroutine 245588
        /home/runner/work/chainloop/chainloop/app/artifact-cas/internal/service/bytestream.go:100 +0x6b8
```

The error happens in this line https://github.com/chainloop-dev/chainloop/blob/4b8af6ec5f0c00e3c7b86ece449e1176e6d1c5e8/app/artifact-cas/internal/service/bytestream.go#L196

and it's likely to happen because the client timeouts or looses connection. Canceling the connection also causes the pod to fail

--- a/app/artifact-cas/internal/service/bytestream.go
+++ b/app/artifact-cas/internal/service/bytestream.go
@@ -34,6 +34,8 @@ import (
 	kerrors "github.com/go-kratos/kratos/v2/errors"
 	"github.com/go-kratos/kratos/v2/log"
 	"google.golang.org/genproto/googleapis/bytestream"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
 // Implements the bytestream interface
@@ -109,7 +111,7 @@ func (s *ByteStreamService) Write(stream bytestream.ByteStream_WriteServer) erro
 
 	// Now it's time to check if the data provider has sent an error
 	if err != nil {
-		if errors.Is(err, context.Canceled) {
+		if errors.Is(err, context.Canceled) || status.Code(err) == codes.Canceled {
 			s.log.Infow("msg", "upload canceled", "digest", req.resource.Digest, "name", req.resource.FileName)
 			return nil
 		}
@@ -192,11 +194,16 @@ func bufferStream(ctx context.Context, stream bytestream.ByteStream_WriteServer,
 		default:
 			// Extract the next chunk of data from the stream request
 			req, err := getWriteRequest(stream)
-			// There is nothing else to read or the client has marked the upload as finished
-			if errors.Is(err, io.EOF) || req.GetFinishWrite() {
+			if err != nil {
+				// If we have finished reading the stream we don't consider it a real error
+				if !errors.Is(err, io.EOF) {
+					bufferErr = err
+				}
 				return
-			} else if err != nil { // If there is any other error we return it
-				bufferErr = err
+			}
+
+			// Check if the client has finished sending data
+			if req.GetFinishWrite() {
 				return
 			}
 

